### ISSUE TITLE
Command cmd P no longer finds files

### ISSUE THREAD

**Issue by joshunger** [Microsoft/vscode#48771]
Issue Type: <b>Bug</b>

Command P (cmd p) no longer finds file / or indexes them correctly?  In the picture below `chrome-drive.js` should appear in the results.

![image](https://user-images.githubusercontent.com/2301847/39319385-2a3aef08-493e-11e8-9263-a42166314c9e.png)

VS Code version: Code - Insiders 1.23.0-insider (b845ce918f2689344795e845ae6ef691916c3079, 2018-04-24T05:10:57.500Z)
OS version: Darwin x64 17.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz (4 x 3100)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.02GB free)|
|Process Argv|/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron ./|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (11)</summary>

Extension|Author (truncated)|Version
---|---|---
sort-lines|Tyr|1.6.1
vscode-eslint|dba|1.4.8
prettier-vscode|esb|1.3.1
flow-for-vscode|flo|0.8.0
python|ms-|2018.3.1
java|red|0.23.0
vscode-fileutils|sle|2.8.1
vscode-java-debug|vsc|0.9.0
vscode-java-pack|vsc|0.3.0
vscode-java-test|vsc|0.5.0
vscode-maven|vsc|0.7.0

</details>

**Comment by roblourens** [comment]
- Did it work before?
- Are the missing results covered by a .gitignore or pattern from `search.exclude` or `files.exclude`?
- Does it happen in every folder, or just one particular folder?

**Comment by joshunger** [comment]
It works intermittently on projects.  On the projects that don't work, it appears to only find files in the root folder.  This might be multiple issues.  Is there a Slack channel I can ask questions to instead of bugging you over issue comments?  In another scenario, it can't find files under a directory called `foo.war`.

**Comment by roblourens** [comment]
Can you show more of the log? Looking for a line that starts with `SearchService#search rg --files ...` and also any errors around that.

Also can you check for errors in the developer tools? (Help > toggle developer tools)

**Comment by joshunger** [comment]
```
[2018-04-30 06:49:12.690] [renderer1] [trace] CommandService#executeCommand workbench.action.quickOpen
[2018-04-30 06:49:12.693] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"","maxResults":0,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:12.718] [renderer1] [debug] SearchService#search rg --files --hidden --case-sensitive -g '!**/.git' -g '!**/.svn' -g '!**/.hg' -g '!**/CVS' -g '!**/.DS_Store' -g '!**/node_modules' -g '!**/bower_components' --no-ignore-parent --follow -- '.'
 - cwd: /Users/junger/dev/devtools-core
 - Sibling clauses: {}
[2018-04-30 06:49:14.240] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"c","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.339] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"ch","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.457] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chr","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.541] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chro","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.614] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrom","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.694] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.849] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:14.966] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-d","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:15.167] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-dr","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:15.343] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-dri","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:15.449] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-driv","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:15.651] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-drive","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:15.794] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-driver","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:15.889] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-driver.","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:16.062] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-driver.j","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
[2018-04-30 06:49:16.168] [renderer1] [trace] SearchService#search {"type":1,"folderQueries":[{"folder":{"$mid":1,"fsPath":"/Users/junger/dev/devtools-core","external":"file:///Users/junger/dev/devtools-core","path":"/Users/junger/dev/devtools-core","scheme":"file"},"excludePattern":{"**/.git":true,"**/.svn":true,"**/.hg":true,"**/CVS":true,"**/.DS_Store":true,"**/node_modules":true,"**/bower_components":true},"fileEncoding":"utf8","disregardIgnoreFiles":false}],"usingSearchPaths":false,"filePattern":"chrome-driver.js","maxResults":512,"sortByScore":true,"cacheKey":"id#50","contentPattern":null,"useRipgrep":true,"disregardIgnoreFiles":false,"ignoreSymlinks":false}
```

**Comment by joshunger** [comment]
![image](https://user-images.githubusercontent.com/2301847/39424064-de545360-4c42-11e8-8ff4-cf6d978cfaf6.png)

**Comment by roblourens** [comment]
Does text search return results when quickopen isn't working?

**Comment by joshunger** [comment]
No, text search does not return results. :(

**Comment by roblourens** [comment]
I'm confused. Can you try invoking the rg command manually in a terminal? From the log, `rg --files --hidden --case-sensitive -g '!**/.git' -g '!**/.svn' -g '!**/.hg' -g '!**/CVS' -g '!**/.DS_Store' -g '!**/node_modules' -g '!**/bower_components' --no-ignore-parent --follow -- '.'`, except change `rg` to the real path to the binary which should be something like `/Applications/Visual\ Studio\ Code\ -\ Insiders.app/Contents/Resources/app/node_modules.asar.unpacked/vscode-ripgrep/bin/rg`

**Comment by roblourens** [comment]
Also, have you tried Stable just to see if there's a difference?

**Comment by joshunger** [comment]
Same issue on stable.

**Comment by roblourens** [comment]
Can you try invoking the rg command manually to see if it errors or produces valid output? See a couple comments up.

**Comment by joshunger** [comment]
```
🐹 $
~/dev/devtools-core
-files --hidden --case-sensitive -g '!**/.git' -g '!**/.svn' -g '!**/.hg' -g '!**/CVS' -g '!**/.DS_Store' -g '!**/node_modules' -g '!**/bower_components' --no-ignore-parent --follow -- '.'
.eslintrc
.editorconfig
.stylelintignore
README.md
.stylelintrc
yarn.lock
.gitignore
package.json
circle.yml
.eslintignore
.flowconfig
lerna.json
docs/versioning.md
docs/local-development.md
flow-typed/debugger-html.js
.github/PULL_REQUEST_TEMPLATE
~/dev/devtools-core
🐷 $
```

**Comment by roblourens** [comment]
So it's missing everything under packages? Can you try simplifying the arguments or throwing in a --no-ignore just to make sure it didn't end up in your .gitignore?

Can you get files that it does return in quickopen?

**Comment by roblourens** [comment]
So that's interesting because this is exactly what the `--no-ignore-parent` flag is supposed to prevent. It should keep rg from looking upwards for gitignore files. Can you try to narrow down the rg command to see if it's simply not working on your machine?

I set up this exact scenario with the same .gitignore file and can't repro it.

**Comment by joshunger** [comment]
I ran out of time to help this week sorry @roblourens

**Comment by roblourens** [comment]
@brocoli open a new issue if we need to investigate. You can disable .gitignore checking with `"search.useIgnoreFiles": false`.

**Comment by joshunger** [comment]
@roblourens do you personally know vscodebot?  He closed the issue. ;)  Do you want to keep this closed?

**Comment by roblourens** [comment]
If you have any time to experiment with running that rg command to figure out why `--no-ignore-parent` isn't working in this case, I'd appreciate it and will keep this open. It's a strange bug that nobody else has reported so I can't do much on my end.

And if the bug just seems to be that flag not working, you could run the rg command with `--debug` and open an issue upstream on https://github.com/BurntSushi/ripgrep.

**Comment by joshunger** [comment]
@roblourens I logged https://github.com/BurntSushi/ripgrep/issues/934

**Comment by roblourens** [comment]
Thanks!

**Comment by roblourens** [comment]
tldr: ripgrep was using the global config of `core.excludesfile=~/.gitignore`.

Now `--no-ignore-global` exists to prevent it from checking that config option.

**Comment by roblourens** [comment]
Verification:

- Set a global gitignore file like `git config --global core.excludesfile ~/.gitignore_global`
- Add a pattern to that global gitignore file that matches some files in your workspace
- Do a text search in vscode and verify that the global gitignore is not applied you can still find matches in those files

**Comment by chrmarti** [comment]
Still an issue with QuickOpen, we need to add that argument there too.

**Comment by chrmarti** [comment]
Maybe also include `--no-config`.

**Comment by roblourens** [comment]
Good catch, thanks!
